examples/pydantic-ai/provider.py

```
@@ -0,0 +1,44 @@
+) -> Dict[str, Any]:
+    """Main provider function for PydanticAI weather agent."""
+    try:
+        config = options.get("config", {})
+        model = config.get("model", "openai:gpt-4o-mini")
+
+        result = asyncio.run(run_weather_agent(prompt, model))
+        output_dict = result.model_dump() if hasattr(result, "model_dump") else result
+
+        return {"output": output_dict}
+
+    except Exception as e:
```

Ruff (0.11.9)

Do not catch blind exception: `Exception`

(BLE001)